src/aggregate/aggregate_exec.c

```diff
@@ -538,8 +538,8 @@ static int execCommandCommon(RedisModuleCtx *ctx, RedisModuleString **argv, int
     goto error;
   }
 
-  SET_DIALECT(r->sctx->spec->used_dialects, r->dialectVersion);
-  SET_DIALECT(RSGlobalConfig.used_dialects, r->dialectVersion);
+  SET_DIALECT(r->sctx->spec->used_dialects, r->reqConfig.defaultDialectVersion);
```

Comment:
does this variable hold only the default dialect or the query dialect? if it can hold the current dialect (given explicitly and not by default), consider dropping the "default" from the name